rsconnect/main.py

```diff
@@ -1393,25 +1396,25 @@ def deploy_app(
         api_key: str,
         insecure: bool,
         cacert: str,
-        entrypoint,
-        exclude,
+        entrypoint: Optional[str],
+        exclude: Optional[list[str]],
         new: bool,
         app_id: str,
         title: str,
-        python,
+        python: Optional[str],
         force_generate: bool,
         verbose: int,
-        directory,
-        extra_files,
-        visibility: typing.Optional[str],
-        env_vars: typing.Dict[str, str],
+        directory: str,
+        extra_files: tuple[str, ...] | list[str],
```

Comment:
This is one spot where the type checker made things a bit more difficult. There's this line below:

```py
extra_files = validate_extra_files(directory, extra_files)
```

But that function returns a list instead of a tuple, and if we have `extra_files: tuple[str, ...]`, the type checker doesn't like it. But maybe I can change that function to return a tuple instead of a list -- I'll check if that looks safe from other places where that function is called.

Reply:
/